endpoint/aurora.go

```diff
@@ -0,0 +1,69 @@
+package endpoint
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/aurora-is-near/relayer2-base/rpc"
+	"github.com/aurora-is-near/relayer2-base/types/primitives"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/valyala/fasthttp"
+)
+
+const (
+	cacheMaxPriorityFeePerGas = 10 * time.Second
+	maxPriorityFeePerGasBody  = `{"id":1,"jsonrpc":"2.0","method":"eth_maxPriorityFeePerGas"}`
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+
+type AuroraClient interface {
+	MaxPriorityFeePerGas() (*primitives.Quantity, error)
+}
+
+type RPCResponse struct {
+	Result *primitives.Quantity `json:"result"`
+}
+
+type AuroraRPC struct {
+	maxPriorityFeePerGasCache      *primitives.Quantity
+	maxPriorityFeePerGasValidUntil time.Time
+	url                            string
+}
+
+func NewAuroraRPC(url string) *AuroraRPC {
+	return &AuroraRPC{url: url}
+}
+
+func (a *AuroraRPC) MaxPriorityFeePerGas() (*primitives.Quantity, error) {
+	now := time.Now()
+
+	if a.maxPriorityFeePerGasCache != nil && a.maxPriorityFeePerGasValidUntil.Before(now) {
```

Comment:
That can be called concurrently so there can be a race.
Mutex should work fine there (atomic is also doable, but a bit more complex and no big need)